### PR TITLE
fix cygwin/bash detection and other issues

### LINKED ISSUE #3: Support :default platform
Could you add a feature where you can specify `:default`, meaning "all other platforms".
e.g.
```
"foo": "run-script-os",
"foo:win32": "echo windows",
"foo:darwin": "echo mac",
"foo:default": "echo everything else",
```

### LINKED ISSUE #11: Use windows when running from Cmder
Based on [this PR](https://github.com/charlesguse/run-script-os/issues/7), it will assume linux when running on cygwin. 

But there are cases where you are running cygwin without the bash syntax, and will cause some issues, for example:
error

I've created a quick check to detect if Cmder is installed in which case will assume Windows (checking if a ENV var declared by Cmder exists) and it seems to be working:
working

I will submit a PR if you wish to include it in your codebase

### LINKED ISSUE #13: Doesn't support install with yarn
`package.json`
```json
{
  "scripts": {
    "preinstall": "run-script-os",
    "install:win32": "echo 'win32'",
    "install:darwin:linux": "echo 'darwin/linux'"
  }
}
```

Running: `yarn` (on its own) is equivalent to running `yarn install`, but fails with the following error:
```bash
yarn install v1.13.0
$ run-script-os
npm ERR! missing script: undefined:darwin
```

While: `yarn install` works correctly:
```bash
yarn install v1.13.0
$ run-script-os

> app@1.0.0 install:darwin:linux /Users/Seb/Development/app
> echo 'darwin/linux'

darwin/linux
[1/5] 🔍  Validating package.json...
...
```

### PR DESCRIPTION
**Pull request by mathiscode**
Fixes #11 
Fixes #13 
Fixes #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,16 @@ del whatever you want in Windows 32/64
...
 ```
 
+### Override detection settings for linux-based shells on Windows
+
+By default, run-script-os will detect cygwin/git bash as Windows. If you would rather your platform be detected as Linux under these environments:
+
+Set environment variable:
+
+```
+RUN_OS_WINBASH=true
```
Comment (GordonSmith) [inline README.md]: Minor:  My instinct when I saw this name was that it did the opposite....  Suggest:  `RUN_OS_WINBASH_IS_LINUX`

Otherwise looks good...

Comment (mathiscode) [reply README.md]: Great point, I've made that change.

**Comment by mathiscode** [comment]
@charlesguse are you still maintaining this project?

**Comment by charlesguse** [comment]
Hey @mathiscode, I am still maintaining this project. I will look through all of the recent activity soon and make sure to get it all merged in.

Thanks!

**Comment by mathiscode** [comment]
Awesome, thanks again for such a useful tool! :smile:

**Comment by phiggins42** [comment]
quick +1, hitting this with :win32 running under git-bash, detects as linux but behaves like windows.  eg:  `FOO=true && npm run bar` needs to be `set FOO=true&&` ...

**Comment by mathiscode** [comment]
> 
> 
> quick +1, hitting this with :win32 running under git-bash, detects as linux but behaves like windows. eg: `FOO=true && npm run bar` needs to be `set FOO=true&&` ...

@phiggins42 While you're waiting for this to be fixed, I would recommend looking at [cross-env](https://www.npmjs.com/package/cross-env), which addresses your specific issue.

**Comment by charlesguse** [comment]
Thanks for your work on this @mathiscode. I like how you fixed my code where I had branching logic for creating the scripts constant!

Sorry for the delay. I need to get some automated testing in place to make this process a lot smoother.

**Comment by mathiscode** [comment]
Happy to help! When I get some more time, I may dig in and help you with the testing. :smile:

**Pull request merged**